File: language/src/main/java/com/deepbeginnings/coiffure/RootNode.java
package com.deepbeginnings.coiffure;

import com.oracle.truffle.api.frame.FrameDescriptor;
import com.oracle.truffle.api.frame.VirtualFrame;

final class RootNode extends com.oracle.truffle.api.nodes.RootNode {
    @Child
    Expr body;

    public RootNode(Language lang, FrameDescriptor locals, Expr body) {
        super(lang, locals);
        this.body = body;
    }

    @Override
    public Object execute(VirtualFrame frame) { return body.execute(frame); }
}

File: src/main/kotlin/RootNode.kt
import com.oracle.truffle.api.frame.FrameDescriptor
import com.oracle.truffle.api.frame.VirtualFrame
import com.oracle.truffle.api.nodes.RootNode as TruffleRootNode

class RootNode(lang: Coiffure, @Child val body: Expr) : TruffleRootNode(lang) {
    override fun execute(frame: VirtualFrame): Any? = body.eval(frame)
}
